code/ResponsiveImageExtension.php

```diff
@@ -1,197 +1,219 @@
-	 */
-	public function allMethodNames()
-	{
-		$methods = array ('createresponsiveset');
-		return array_merge($methods, $this->getResponsiveSets());
-	}
-
-
+    /**
+     * @var array
+     * @config
+     */
+    private static $default_args = array(800, 600);
+
+    /**
+     * @var string
+     * @config
+     */
+    private static $default_method = 'SetWidth';
+
+    /**
+     * @var boolean
+     * @config
+     */
+    private static $htmleditorfield_srcset = true;
+
+    /**
+     * @var array
+     * @config
+     */
+    private static $htmleditorfield_srcset_densities = array(1, 2);
+
+    /**
+     * @var array A cached copy of the image sets
+     */
+    protected $configSets;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->configSets = Config::inst()->get(__CLASS__, 'sets');
+    }
+
+    /**
+     * A wildcard method for handling responsive sets as template functions,
+     * e.g. $MyImage.ResponsiveSet1
+     *
+     * @param string $method The method called
+     * @param array $args The arguments passed to the method
+     * @return HTMLText
+     */
+    public function __call($method, $args)
+    {
+        if ($config = $this->getConfigForSet($method)) {
+            return $this->createResponsiveSet($config, $args, $method);
+        }
+    }
+
+    /**
+     * Requires the necessary JS and sends the required HTML structure to the
+     * template for a responsive image set.
+     *
+     * @param array $config The configuration of the responsive image set
+     * @param array $defaultArgs The arguments passed to the responsive image
+     *                           method call, e.g. $MyImage.ResponsiveSet(800x600)
+     * @param string $set The method, or responsive image set, to generate
+     * @return SSViewer
+     */
+    protected function createResponsiveSet($config, $defaultArgs, $set)
+    {
+        Requirements::javascript(RESPONSIVE_IMAGES_DIR . '/javascript/picturefill/picturefill.min.js');
+
+        if (!isset($config['sizes']) || !is_array($config['sizes'])) {
+            throw new Exception("Responsive set $set does not have sizes defined in its config.");
+        }
+
+        if (empty($defaultArgs)) {
+            if (isset($config['default_args'])) {
+                $defaultArgs = $config['default_args'];
+            } else {
+                $defaultArgs = Config::inst()->get(__CLASS__, 'default_args');
+            }
+        }
+
+        if (isset($config['method'])) {
+            $methodName = $config['method'];
+        } else {
+            $methodName = Config::inst()->get(__CLASS__, 'default_method');
+        }
+
+        $sizes = ArrayList::create();
+        foreach ($config['sizes'] as $query => $args) {
+            if (is_numeric($query) || !$query) {
+                throw new Exception("Responsive set $set has an empty media query defined.
+					Please check your config format");
```

Comment:
On one line it exceeds 120 characters (PSR-2) which is why I broke it onto two lines here. I suppose I could re-phrase the message instead...